### PR TITLE
Support response_format type=json_object

### LINKED ISSUE #1171: Support response_format type=json_object
## Summary

`response_format: {"type": "json_object"}` is still not accepted by the OpenAI-compatible server endpoints. The server currently supports `json_schema` structured outputs from #1047, but requests using the older OpenAI JSON mode value fail with `Unsupported response_format type: 'json_object'`.

## Background

Related prior work:

- #1047 added `json_schema` support and explicitly left `json_object` for future work.
- #993 and #959 attempted JSON mode support, but both were closed unmerged.
- #561 requested structured output support and was closed after #1047.

## Proposed behavior

Accept `json_object` in both OpenAI-compatible input shapes:

- Chat Completions: `response_format: {"type": "json_object"}`
- Responses API: `text: {"format": {"type": "json_object"}}`

Because the server now has llguidance-backed structured generation, `json_object` can be implemented as a permissive JSON object schema:

```json
{"type": "object"}
```

This preserves the OpenAI compatibility surface while avoiding the prompt-injection/whitespace failure mode of instruction-only JSON mode. It also keeps `json_schema` as the recommended path when callers need an explicit schema.

## Scope

- Accept `json_object` as a valid response format.
- Route it through the existing structured logits processor path.
- Cover Chat Completions and Responses API request shapes.
- Keep the existing limitation that structured outputs are not supported with speculative decoding.

## References

OpenAI documents `json_object` as older JSON mode and recommends `json_schema` for schema-constrained outputs: https://platform.openai.com/docs/guides/structured-outputs?api-mode=chat

### PR DESCRIPTION
**Pull request by eloe**
## Summary

Adds OpenAI-compatible `response_format: {"type": "json_object"}` support to the server.

`json_object` is routed through the existing llguidance-backed structured output path as a permissive JSON object schema:

```json
{"type": "object"}
```

This preserves compatibility with older OpenAI JSON mode while using the server's existing constrained generation machinery instead of prompt injection.

Fixes #1171.

## Details

Supported request shapes:

- Chat Completions: `response_format: {"type": "json_object"}`
- Responses API: `text: {"format": {"type": "json_object"}}`

The existing `json_schema` path remains unchanged and should still be preferred when callers need a specific schema. Existing speculative decoding behavior is unchanged: structured response formats are rejected for speculative decoding.

## Validation

```text
uv run --with pytest pytest mlx_vlm/tests/test_server.py -q
83 passed, 2 warnings
```

## Related Work

- #1047 added `json_schema` support and left `json_object` out of scope.
- #993 and #959 attempted JSON mode support but were closed unmerged.
- #561 requested structured output support and was closed after #1047.

## Note

Replaces closed draft #1172 after renaming the fork branch to remove the old prefix.

### CODE REVIEW

**Comment by lucasnewman** [comment]
@eloe Thank you! We're fixing this here since this one has drifted a bit: https://github.com/Blaizzy/mlx-vlm/pull/1299
